amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/admin/Queues.java

```diff
@@ -77,13 +82,19 @@ public void getQueue(@Suspended final AsyncResponse response,
     @PUT
     @Path("/{vhost}/{queue}")
     public void declareQueue(@Suspended final AsyncResponse response,
-                                @PathParam("vhost") String vhost,
-                                @PathParam("queue") String queue,
-                                QueueDeclareParams params) {
-        declareQueueAsync(vhost, queue, params)
+                             @PathParam("vhost") String vhost,
+                             @PathParam("queue") String queue,
+                             QueueDeclareParams params,
+                             @QueryParam("authoritative") @DefaultValue("false") boolean authoritative) {
+        TopicName topicName = TopicName.get(TopicDomain.persistent.toString(),
+                NamespaceName.get("public", vhost), PersistentQueue.TOPIC_PREFIX + queue);
+        validateTopicOwnershipAsync(topicName, authoritative)
+                .thenCompose(__ -> declareQueueAsync(vhost, queue, params))
```

Comment:
I think we should pass the params all the way to the PersistentQueue.

Reply:
fixed